apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/app/BlockchainProcessorImpl.java

```diff
@@ -480,6 +487,7 @@ public void processPeerBlock(JSONObject request) throws AplException {
     public void fullReset() {
         globalSync.writeLock();
         try {
+            lookupDexOrderProcessor().suspendContractProcessor();
```

Comment:
Not the best idea, try to decouple exchange and related classes from the core blockchain logic

Reply:
> Not the best idea, try to decouple exchange and related classes
Try explain idea... E.g. I don't understand. do you mean Inject DexOrderProcessor via BlockchainProcessorImpl constructor ?